cmd/cmd.go

```diff
@@ -60,23 +62,35 @@ var DownloadCmd = &cobra.Command{
 		if verbose {
 			logger.SetLevel(log.DebugLevel)
 		}
-		switch target.Distro.Display {
-		case "Fedora", "RHEL":
-			backend, err = rpm.NewRedHatBackend(&target, rpmReposDir, logger)
-		case "CentOS":
-			backend, err = rpm.NewCentOSBackend(&target, rpmReposDir, logger)
-		case "openSUSE":
-			backend, err = rpm.NewOpenSUSEBackend(&target, rpmReposDir, logger)
-		case "SLE":
-			backend, err = rpm.NewSLESBackend(&target, rpmReposDir, logger)
-		case "Debian", "Ubuntu":
+		switch target.Distro.Family {
```

Comment:
Look like `target.Distro.Family` isn't a value which we can always expect be set (at least, according to the failing `opensuse` test). I think it would be better to rely exclusively on `target.Distro.Display`

Reply:
`target.DistroFamily` is indeed inferred from `target.Distro.Display` but if we don't have `Family` set, it means that it's an unknown `Distro` that we'll have to handle in the `switch... case` and that's probably better if it's the one in gopsutil

Reply:
`gopsutil` currently sets `Distro` but leaves `Family` blank when the distribution is `"opensuse-leap", "opensuse-tumbleweed", "opensuse-tumbleweed-kubic, "sles", "sled", "caasp"` or `"cos"`.

We can certainly patch `gopsutil` so that it infers `Family` more accurately, but at the moment we have various tests failing because of this issue, and leaving the outer `switch` statement so that it checks `Display` instead of `Family` is an easy fix.